internal/serialization/big_int.go

```diff
@@ -0,0 +1,293 @@
+/*
```

Comment:
Honestly, the code is too complex to review but I will assume it is correct, given that we do have lots of tests covering edge cases, small and large value :+1:

Reply:
:pray: